pandas/io/parsers/readers.py

```diff
@@ -1515,6 +1515,11 @@ def _clean_options(
             if engine == "c" and sep == r"\s+":
                 result["delim_whitespace"] = True
                 del result["delimiter"]
+            elif engine == "pyarrow" and (sep in ("\\s+", "  ", "||")):
```

Comment:
For the test adding those examples is great, but here it makes more sense to simply check that `len(sep) > 1`

Reply:
You 're right, and since len(sep) > 1 is already checked in the outer if statement, all i have to do is remove the sep in ().